### PR TITLE
fix(evm): update the dependency evmos to v16

### PR DESCRIPTION
**Pull request by TimmyExogenous**
## Description

This PR updates the dependency `evmos` to v16 to fix the gas estimate error.

This PR temporarily disables the price feeder to fix the issue caused by two different versions of the EVMOS dependency.
It needs to be re-enabled after the price feeder updates the EVMOS dependency to v16 and updates the Exocore dependency to the version that includes this fix.

Because most of the changes are the imported version, please review the following files:
* app/app.go
* x/evm
* precompiles/assets/assets.go
* precompiles/avs/avs.go
* precompiles/delegation/delegation.go
* precompiles/reward/reward.go
* precompiles/slash/slash.go
* proto/exocore/assets/v1/tx.proto

### CODE REVIEW

**Comment by TimmyExogenous** [comment]
The lint error regarding vulnerabilities should be ignored. Because the first vulnerability has been patched, and the second is related to ICS20 which has been disabled in the active precompile list.

**Comment by MaxMustermann2** [comment]
Can you please list out the files we should review? There are too many cosmetic changes from `v14` to `v16`,

**Comment by TimmyExogenous** [comment]
> Can you please list out the files we should review? There are too many cosmetic changes from `v14` to `v16`,

updated in the description

**Review by bwhour — APPROVED** [review]
looks good.

**Comment by TimmyExogenous** [comment]
This PR temporarily disables the price feeder to fix the issue caused by two different versions of the EVMOS dependency.
It needs to be re-enabled after the price feeder updates the EVMOS dependency to v16 and updates the Exocore dependency to the version that includes this fix. @cloud8little @leonz789

File: x/evm/keeper/precompiles.go [hunk 1]
```diff
@@ -135,3 +137,65 @@
...
+// Check https://stackoverflow.com/questions/21362950/getting-a-slice-of-keys-from-a-map.
+func (k Keeper) GetAvailablePrecompileAddrs() []common.Address {
+	addresses := make([]common.Address, len(k.precompiles))
+	i := 0
+
+	//#nosec G705 -- two operations in for loop here are fine
+	for address := range k.precompiles {
+		addresses[i] = address
+		i++
+	}
```
[the remark this answers is not in the document]
Comment (TimmyExogenous) [reply x/evm/keeper/precompiles.go:194]: The addresses list will be sorted using the following code.

**Review by trestinlsd — APPROVED** [review]

**Review by leonz789 — APPROVED** [review]

**Pull request merged**